runtime/parsing.js

```diff
@@ -73,8 +76,27 @@ function caml_parse_engine(tables, env, cmd, arg)
   var tbl_table = 12;
   var tbl_check = 13;
   // var _tbl_error_function = 14;
-  // var _tbl_names_const = 15;
-  // var _tbl_names_block = 16;
+  var tbl_names_const = 15;
+  var tbl_names_block = 16;
+
+  function token_name(names, number)
+  {
+    var str = caml_jsstring_of_string(names);
+    if (str[0] == '\x00')
+      return "<unknown token>";
+    return str.split('\x00')[number];
+  }
+
+  function print_token(tok)
+  {
+    var token;
+    if (tok instanceof Array) {
+      token = token_name(tables[tbl_names_block], tok[0]);
+    } else {
+      token = token_name(tables[tbl_names_const], tok);
+    }
+    console.log("State", state, ": read token", token);
```

Comment:
The native runtime prints some additional information. Could we try to replicate the behavior.

Can we add an expect test for the trace feature ?

Reply:
I've pushed a commit to do this.